bot/src/main/kotlin/io/heapy/kotbot/bot/rule/DevRules.kt

```diff
@@ -0,0 +1,31 @@
+package io.heapy.kotbot.bot.rule
+
+import io.heapy.kotbot.bot.BotStore
+import org.telegram.telegrambots.meta.api.objects.Update
+
+class GetIdRule(private val store: BotStore) : Rule {
```

Comment:
Maybe introduce another interface for commands?
`Command` (you can choose better name if any) interface can know about command name, and maybe params, so boilerplate like this:

```
    if(!update.hasMessage()) return emptyList()
        val message = update.message
        if(!message.hasText()) return emptyList()
        val text = message.text
        val chat = message.chat
```

can be avoided.

Reply:
Okay, so I may have gone in a little different direction with it than planned at the beginning. Playing with a composition I've based different kinds of rules on high-order functions and not on hierarchy of `Rule` derivatives. To me it seems a bit more extensible and a bit less verbose. Let me know if you are okay with that or not.